pkgs/development/perl-modules/generic/default.nix

```diff
@@ -46,6 +46,10 @@ toPerlModule(stdenv.mkDerivation (
     builder = ./builder.sh;
     buildInputs = buildInputs ++ [ perl ];
     nativeBuildInputs = nativeBuildInputs ++ [ (perl.dev or perl) ];
+    
+    # environment variables passed to ./builder.sh
+    hostperl = perl;
+    devperl = perl.dev or perl;
```

Comment:
`lib.getDev` ?

Reply:
Actually, I'd rename this `.dev`.
It is not `dev` in the sense as there is `cairo.dev`, it is a hack used only for cross-compilation of perl packages.
In that `.dev`, there is no pkg-config and include files as one might expect, but a `buildPlatform`'s `perl` binary which uses `hostPlatform`'s libraries.
So far they are pure Perl and there is no `.so`, everything is fine.

Insisting that it is `.dev` up using lib functions for that is misleading.